### DISCUSSION TITLE
NxNandManager_v5.1

### DISCUSSION THREAD

**Discussion by eliboa** [eliboa/NxNandManager#76] [General]
The following was changed since the last release:

## Bug fixes:
Issue| Resolution
---- | ----------- 
#74, #37 | NAND without GPT backup is now recognized as a valid NxStorage 
#71 | A bug was fixed that caused checksum mismatch error
#61, #47 | Several bugs were fixed regarding partition formatting
#59 | Filebased emunand with extra dots in file path were not recognized, fixed
#57, #44 | File paths with non ASCII characters are now properly managed
#46 | Mariko: A bug was fixed that prevented DeviceID from being recognized
#24 | A bug was fixed that caused crypto validation error for empty/formatted partitions

### Partition mounting (virtual disk / virtual FS): 
- New option: User can select mouting point (drive letter)
- New option: Nintendo's archive folder virtualization (handled as single file NCA, file size can exceed 4 GB) 
- New option: Partition can be mounted as read-only
- Virtual FS now uses multithreading which allows for file sharing & faster transfer speeds.

![Png](https://eliboa.com/nxnm_5.1_mount_dialog.png)

### NAND partition formatting (USER, SYSTEM): 
Code has been redesigned. FatFS lib is now used.

### Explorer (Beta):
- Explore the content of USER & SYSTEM partitions. Provides useful info: title name & id, user name & id, etc.
- List, save or extract files from saveFS (title saves, under /save folder)
- List, save, decrypt or extract content from NCAs (installed titles) using hactoolnet

![Png](https://eliboa.com/nxnm_5.1_explorer_dialog_01.png)   
![Png](https://eliboa.com/nxnm_5.1_explorer_dialog_02.png)   
![Png](https://eliboa.com/nxnm_5.1_explorer_dialog_03.png)   

#### Please note:
- You'll need to re-configure your keyset (import all keys from both prod.keys & title.keys).
- Content (JPEG) will be downloaded from Nintendo's public CDN and cached in ```/cache``` folder.
- Content will be downloaded from eliboa.com (nca filename DB as json file, title DB as json file).

### Keyset configuration: 
You can import all sorts of keys into keyset, not only Bis keys (useful for Explorer).

![Png](https://eliboa.com/nxnm_5.1_keyset_dialog.png)
### Firmware detection:
- BOOT0, SYSTEM: added support for latest firmwares (up to 13.2.1)
- SYSTEM: added a feature to query firmware version from a remote database : eliboa.com/switch/nca.php (#75)

![GitHub Releases (by Release)](https://img.shields.io/github/downloads/eliboa/NxNandManager/v5.1/total)

<hr /><em>This discussion was created from the release <a href='https://github.com/eliboa/NxNandManager/releases/tag/v5.1'>NxNandManager_v5.1</a>.</em>

**Comment by OrGoN3** [comment]
You sneaky fox. Thanks for all of the new additions. Am I correct in that you're formatting fat32 cards with 16k (0x4000) cluster size? Can you increase this to 64k (0x10000)?  Is it as simple as just changing CLUSTER_SIZE or will that affect anything else negatively? Yours is one of the few tools that will create a hidden emu partition, instead of me having to manually do it. I'd just use hekate but got some reason I'm sold on hidden emu partition. No idea why. Anyway,

Thanks eliboa!

**Reply by eliboa** [reply]
CLUSTER_SIZE is a preprocessor directive used everywhere in code, not only in emunand creation. Initially it described the cluster size of internal FAT32 Bis partitions. If you change it, i'm pretty sure the program would just crash when doing crypto on BIS partitions. With minor changes in emunand creation function I would be able to change cluster size but the proper way to achieve this is to redesign code to implement FatFS external lib. This library is already used for mounting & Explorer and it would also support exFAT formatting. Why miss an opportunity to kill two birds with one stone 😉 ?

Added to my TODO list.

**Comment by Interloper101** [comment]
Does this still work on 14.0.0?

**Comment by redretro** [comment]
I have an edev switch, can I modify the nand to be able to use it as a normal switch, if so how?
thkx
